### PR TITLE
Refactor aria live announcer to use `aria-relevant: additions` attribute with GC mechanism

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal (ui):  Refactor aria live announcer to use `aria-relevant: additions` attribute with GC mechanism
MINOR BREAKING CHANGE (ui): Drop region name argument from `AriaLiveAnnouncer#announce`  method. 

---

### Additional information

Part of https://github.com/ckeditor/ckeditor5/pull/16197
Part of https://github.com/cksource/ckeditor5-commercial/pull/6166

### CODE REVIEW

File: packages/ckeditor5-ui/src/arialiveannouncer.ts [hunk 1]
```diff
@@ -166,58 +144,104 @@ export class AriaLiveAnnouncerView extends View {
...
+	private get _listElement(): HTMLElement | null {
+		return this.element!.querySelector( 'ul' )!;
 	}
 }
 
 type AriaLiveAnnouncerPolitenessValue = typeof AriaLiveAnnouncerPoliteness[ keyof typeof AriaLiveAnnouncerPoliteness ];
 
+type AriaLiveAppendContentAttributes = {
+	announcement: string;
+	isUnsafeHTML?: boolean;
```
Comment (oleq) [inline packages/ckeditor5-ui/src/arialiveannouncer.ts:241]: Let's get rid of this flag as it seems it won't be used anyway.

**Review by oleq — COMMENTED** [review]
This PR will be fine when we remove an obsolete flag.

File: packages/ckeditor5-ui/src/arialiveannouncer.ts (hunk 1 above)
Comment (Mati365) [reply packages/ckeditor5-ui/src/arialiveannouncer.ts:241]: It is used in destination branch https://github.com/cksource/ckeditor5-commercial/pull/6163/files#diff-c3e2cfc5f2b64e23fc4bb25cdb3e76e1615a85a1fd0fac616a71d2271cb1456cR491-R494

**Comment by Mati365** [comment]
@oleq This flag is used in AI integration.

File: docs/umberto.json [hunk 2]
```diff
@@ -299,6 +299,7 @@
 		"updating/migration-from-ckeditor-4.html#migration-from-ckeditor-4-faq": "updating/ckeditor4/ckeditor4-troubleshooting.html",
 		"updating/migration-from-ckeditor-4.html#plugins-compatibility-table": "updating/ckeditor4/ckeditor4-plugin-compatibility.html",
 		"updating/migration-from-ckeditor-4.html#configuration-options-compatibility-table": "updating/ckeditor4/ckeditor4-configuration-compatibility.html",
+		"updating/migration-from-cke4.html#configuration-options-compatibility-table": "updating/ckeditor4/ckeditor4-configuration-compatibility.html",
```
Comment (oleq) [inline docs/umberto.json:302]: How did this change get in here?

**Review by oleq — CHANGES_REQUESTED** [review]
[CI fails](https://app.circleci.com/pipelines/github/ckeditor/ckeditor5/4657/workflows/2a436fe5-31bc-430b-8559-e07e47a730b5/jobs/26283?invite=true#step-114-1682_122) (also locally) and I noticed that unit tests pass if I early return from `AriaLiveAnnouncer#constructor()`. There's something odd going on there, some kind of clash?

The functionality works great, though. I tested it in NVDA, JAWS, and VoiceOver and the simplified approach works out 🙂

File: docs/umberto.json (hunk 2 above)
Comment (Mati365) [reply docs/umberto.json:302]: Don't know. Probably wrong auto merge. Will be changed

**Pull request merged**